File: 2017 COMLAN/Timer/src-weaved/matlab/foobar.m
function foobar()
   tic;
   foo();
   matisse_time0 = toc;
end

function [a] = foo()
   a = 0;
   for i = 0:5
      tic;
      a = a + bar();
      matisse_time1 = toc;
   end
end

function [result] = bar()
   result = 1.0;
end

File: 2017 COMLAN/Timer/src-weaved/java/FooBar.java



public class FooBar {
    public double bar() {
        return 1.0;
    }

    public double foo() {
        double a = 0;
        for (int i = 0; i < 5; i++) {
            long kadabra_timing_start_0 = System.nanoTime();
            a += bar();
            double kadabra_timing_interval_0 = (double)(System.nanoTime() - kadabra_timing_start_0) /  (double)1000;
            System.out.printf("Time:%fus\n", kadabra_timing_interval_0);
        }
        return a;
    }

    public static void main(String[] args) {
        long kadabra_timing_start_1 = System.nanoTime();
        new FooBar().foo();
        double kadabra_timing_interval_1 = (double)(System.nanoTime() - kadabra_timing_start_1) /  (double)1000;
        System.out.printf("Time:%fus\n", kadabra_timing_interval_1);
    }
}

